R/bayesianpredictionperformance.R

```diff
@@ -0,0 +1,259 @@
+    model$dependOn(c(.metamiscDependencies, .metamiscDependenciesBayesian))
+    jaspResults[["model"]] <- model
+  } else
+    return()
+
+
+  pars <- list(
+    model.oe    = if (options[["measure"]] == "OE")    options[["linkOE"]],
+    model.cstat = if (options[["measure"]] == "cstat") options[["linkCstat"]]
+  )
+
+  pars$hp.mu.mean <- options[["priorMuNMeam"]]
+  pars$hp.mu.var  <- options[["priorMuNSD"]]^2
+
+  if (options[["priorTau"]] == "priorTauU") {
+    pars$hp.tau.min <- options[["priorTauUMin"]]
+    pars$hp.tau.max <- options[["priorTauUMax"]]
+    pars$hp.tau.dist="dunif"
+  } else if (options[["priorTau"]] == "priorTauT") {
+    pars$hp.tau.dist  <- "dhalft"
+    pars$hp.tau.mean  <- options[["priorTauTLocation"]]
+    pars$hp.tau.sigma <- options[["priorTauTScale"]]
+    pars$hp.tau.df    <- options[["priorTauTDf"]]
+    pars$hp.tau.min   <- options[["priorTauTMin"]]
+    pars$hp.tau.max   <- options[["priorTauTMax"]]
+  }
+
+  dataset <- .metamiscOmitNAs(dataset, options)
+
+  fit <- tryCatch(metamisc::valmeta(
+    measure    = options[["measure"]],
+    cstat      = if (options[["measure"]] == "cstat" && options[["inputMeasure"]] != "")              dataset[, options[["inputMeasure"]]],
+    cstat.se   = if (options[["measure"]] == "cstat" && options[["inputSE"]] != "")                   dataset[, options[["inputSE"]]],
+    cstat.cilb = if (options[["measure"]] == "cstat" && sum(unlist(options[["inputCI"]]) != "") == 2) dataset[, options[["inputCI"]][[1]][1]],
+    cstat.ciub = if (options[["measure"]] == "cstat" && sum(unlist(options[["inputCI"]]) != "") == 2) dataset[, options[["inputCI"]][[1]][2]],
+    OE         = if (options[["measure"]] == "OE" && options[["inputMeasure"]] != "")                 dataset[, options[["inputMeasure"]]],
+    OE.se      = if (options[["measure"]] == "OE" && options[["inputSE"]] != "")                      dataset[, options[["inputSE"]]],
+    OE.cilb    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][1]],
+    OE.ciub    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][2]],
+    N          = if (options[["inputN"]] != "")      dataset[, options[["inputN"]]],
+    O          = if (options[["inputO"]] != "")      dataset[, options[["inputO"]]],
+    E          = if (options[["inputE"]] != "")      dataset[, options[["inputE"]]],
+    slab       = if (options[["inputLabels"]] != "") dataset[, options[["inputLabels"]]],
+    method     = "BAYES",
+    pars       = pars,
+    adapt      = options[["adapt"]],
+    burnin     = options[["burnin"]],
+    sample     = options[["sample"]],
+    n.chains   = options[["chains"]]
+  ),error = function(e)e)
+
+  # error handling
+  if (any(class(fit) %in% c("simpleError", "error"))) {
+    .quitAnalysis(paste0("metamisc package failed with the following error: '", fit[["message"]], "'"))
+  }
```

Comment:
So this is not ideal behavior for the user. Better is to do
```r
container$setError(gettextf("metamisc package failed with the following error: '%s''", fit[["message"]])
```
where `container` is the outermost container of the results.
Also, this cannot be translated, of course, you need to use `gettextf`.

Reply:
hmm, but when the fitting function fails, the rest of the analysis is unusable, so isn't it better to completely stop it?

Reply:
Well, the results will look pretty similar, except that you can maybe still show descriptive information. However, if users click on some options while there is an error they would get feedback in the form of greyed-out tables/ plots.

With this change the other functions become:
```R
table <- createJaspTable(...) # or some plot

table$addColumnInfo(...)

container[[name]] <- table

if (!ready || container$getError()) # <- should be the only change necessary to the other functions
  return()
```

Reply:
ok, changed